### DISCUSSION TITLE
In one of the server, rke2-server restart took 5 hours to come up

### DISCUSSION THREAD

**Discussion by krp2** [rancher/rke2#4310] [Q&A]
**Environmental Info:**
RKE2 Version: 

rke2 version v1.24.7+rke2r1 (5fd5150f752a663efa16ac3ae2aac5d6bba6c364)
go version go1.18.1b7

Node(s) CPU architecture, OS, and Version: 

Linux mno-1 5.3.18-150300.59.106-default #1 SMP Mon Dec 12 13:16:24 UTC 2022 (774239c) x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:

3 servers

**Describe the bug:**
In the first master node, after cluster is being deployed for some days, the rke2-server service has restarted and it has taken around 5 hours to start. We see that containerd is failing to comeup. Can you please help us understand what could be the reason for this?

**Steps To Reproduce:**

- Installed RKE2 with 3 masters
- rke2-server service restarted and failed to come up 

**Expected behavior:**
rke2 service should come up quickly

**Actual behavior:**
rke2 service took 5 hours to come up

**Additional context / logs:**
journaltl logs for rke2-server service and containerd logs are attached. Please check the logs between timeframe "2023-05-22T00:20" to "2023-05-22T05:16"
rke2 service restarted around 2023-05-22T00:20 and finally came up after 5 hours around 2023-05-22T05:16

[rke2-server-journalctl.log.txt](https://github.com/rancher/rke2/files/11615213/rke2-server-journalctl.log.txt)
[containerd.log](https://github.com/rancher/rke2/files/11615296/containerd.log)

**Comment by brandond** [comment]
There's not enough information in the logs to determine what was going on. The successful containerd startup completes in 24 minutes, which is just under the 30 minutes that RKE2 allows for containerd to successfully start:

```
time="2023-05-22T04:52:24.909668511-05:00" level=info msg="Connect containerd service"
time="2023-05-22T04:52:24.909717072-05:00" level=info msg="Get image filesystem path \"/var/lib/rancher/rke2/agent/containerd/io.containerd.snapshotter.v1.overlayfs\""
time="2023-05-22T04:52:24.918499296-05:00" level=info msg="Start subscribing containerd event"
time="2023-05-22T04:52:24.918552367-05:00" level=info msg=serving... address=/run/k3s/containerd/containerd.sock.ttrpc
time="2023-05-22T04:52:24.918584721-05:00" level=info msg="Start recovering state"
time="2023-05-22T04:52:24.918727515-05:00" level=info msg=serving... address=/run/k3s/containerd/containerd.sock
time="2023-05-22T04:52:24.918745663-05:00" level=info msg="containerd successfully booted in 0.299060s"
...
time="2023-05-22T05:16:14.156944132-05:00" level=info msg="Start event monitor"
time="2023-05-22T05:16:14.156985848-05:00" level=info msg="Start snapshots syncer"
time="2023-05-22T05:16:14.156999954-05:00" level=info msg="Start cni network conf syncer for default"
time="2023-05-22T05:16:14.157007285-05:00" level=info msg="Start streaming server"
```

The failed startups never finish scanning the image store, or at least not within the time alloted. The most likely cause of this is high disk latency or other IO contention preventing containerd from scanning the image store in a timely manner. Do you have any other logs or metrics for this host that might cover that time period?

I've honestly never seen containerd take this long to restore state for the image store; I highly suspect something is wrong with your underlying storage.

**Comment by krp2** [comment]
@brandond  Attaching the etcd logs (before rke2 restart got triggered) , output of "top" command, output of "free" command and disk space

[etcd-logs.txt](https://github.com/rancher/rke2/files/11655342/etcd-logs.txt)
[free.txt](https://github.com/rancher/rke2/files/11655344/free.txt)
[mno-1-disk-space.log](https://github.com/rancher/rke2/files/11655345/mno-1-disk-space.log)
[top.txt](https://github.com/rancher/rke2/files/11655346/top.txt)

Can you please let us know if these logs help to identify the issue? Or if you are looking for any particular log please let us know. We will collect it in the next attempt and share it here.

**Reply by krp2** [reply]
@brandond Can you please clarify this?

**Reply by brandond** [reply]
There's nothing in these logs that sheds any more light on the situation. Most of them don't even cover the time period in question, or don't provide timestamps at all (such as the top and free output).

I will say that you have a few system processes using a full core, which might be worth looking into?
```
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
13342 delladm+  20   0  736964  33492   5520 S 108.0 0.103 331:35.74 ./manager --enable-leader-election
17884 root      20   0  244968 111440    488 S 100.0 0.342 220:15.60 /usr/local/bin/td-agent-bit -c /fluent-bit/etc/fluent-bit.conf
 5198 root      20   0  105048  83632  16032 R 96.00 0.257   0:02.49 python3 /snmp_app/snmp_entry.py
```

I don't think that I'm going to be able to tell you what happened. If it happens again, you'd need to log on to the host while containerd is starting up, look at processor utilization and disk IO statistics, and perhaps strace the containerd process to see what it is spending its time doing.
